Add webcron to scheduler config

ditsuke opened the pull request:
### Summary of Changes
Auto-generate the webcron key and enhance scheduler config form.


### Testing Instructions
Test the scheduler component configuration for the webcron key. It should auto-generate on enabling webcron and saving the form. Toggling the reset radio should reset the key on form save.
Webcron base link should be copy-able through the link field in the form, which is visible when the webcron feature is enabled and a key has been generated.

ditsuke committed:
Add webcron to scheduler config

Updates the `com_scheduler` config.xml with the webcron fieldset.

ditsuke committed:
Add webcron key autogen and more to Schedulerunner

- Schedulerunner auto-generates the webcron key and does some form
  usability enhancements on the config form (much like the user API
  token plugin).

ditsuke committed:
Add custom field for webcron link

- Adds new field 'WebcronLinkField', this field is not really needed
  except for to support the custom layout location.
- Allows for the webcron link to be copies on click, much like the user
  API token field.
- Adds field JS to the `com_scheduler` media source provider.

ditsuke committed:
Update com_scheduler language file

Adds new strings for the config form and updates some refs.

ditsuke committed:
Merge 'origin/soc/scheduler/play-button' into webcron-key

Conflicts:
    - plugins/system/schedulerunner/schedulerunner.php

ditsuke committed:
Fix regression in Scheduler class

- Fix regression due to newly protected status
  of Task::snapshot + accessor.
- Update some doc-blocks and related formatting.
- Optimise some imports.
- Minor miscellaneous changes.

bembelimen commented on plugins/system/schedulerunner/schedulerunner.php:
```diff
@@ -183,15 +218,89 @@ public function runTestCron(Event $event)
+			return;
+		}
+
+		$params = new Registry(json_decode($table->params));
```
I think, ```json_decode``` is not needed.

ditsuke replied:
Thanks, it was not!

bembelimen commented on plugins/system/schedulerunner/schedulerunner.php:
```diff
@@ -183,15 +218,89 @@ public function runTestCron(Event $event)
+		[$context, $table] = $event->getArguments();
+
+		if ($context !== 'com_config.component'
+			|| $this->app->input->get('component') !== 'com_scheduler')
```
Did you check, if in ```$table``` the ```component``` is set? Then you could check for that.

ditsuke replied:
Removed the input check and checking only for the `$table->name`.

ditsuke committed:
Fix test run JS

Fixes apparent JS parse error on non-zero task exits.

ditsuke committed:
Improve form manipulation code

- Check for `com_scheduler` in the subject table for table event.
- Minor code simplification.
